tests/testthat/test-variable-catalog.R

```diff
@@ -99,12 +98,38 @@ with_mock_crunch({
                 type=c("numeric", "categorical", "multiple_response")
             )))
     })
+    
+    test_that("Variable Catagory as.data.frame method", {
+        expect_equal(as.data.frame(varcat[1:3]),
+            data.frame(
+                alias = c("birthyr", "gender", "mymrset"),
+                name  = c("Birth Year", "Gender", "mymrset"),
+                type  = c("numeric", "categorical", "multiple_response"),
+                stringsAsFactors = FALSE
+            ))
+    })
+    test_that("as.data.frame method returns all fields when keys = 'all'", {
+        expect_identical(
+            names(as.data.frame(varcat, keys = "all")),
+            c("name", "discarded", "alias", "type", "id", "description", 
+                "notes", "subvariables", "subvariables_catalog", "resolution", 
+                "rollup_resolution")
+        )
+    })
+    test_that("As.data.frame method errors correctly", {
+        expect_error(as.data.frame(varcat[1:3], keys = "Not a field at all"),
+            "'Not a field at all'is an invalid key for catalogs of class VariableCatalog."
```

Comment:
Looks like you're missing a space in your error message. Also, I'd like to see a test where the keys are a mix of valid and invalid entries, like c("name", "banana")